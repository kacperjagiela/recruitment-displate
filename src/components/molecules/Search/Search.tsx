import { CloseIcon } from '@chakra-ui/icons';
import { Input, InputRightElement } from '@chakra-ui/input';
import { Box, Text } from '@chakra-ui/layout';
import { InputGroup } from '@chakra-ui/react';

interface Props {
    searchValue: string;
    onSearch: (value: string) => void;
    clearSearch: () => void;
}

const Search: React.FC<Props> = ({ searchValue, onSearch, clearSearch }: Props) => (
    <Box color="white" textStyle="h2" py={12}>
        <Text as="h2" textStyle="h2">
            Woof Woof Gallery
        </Text>
        <Text as="h3" textStyle="h3" py={2}>
            Already know the bread?
        </Text>
        <Text as="h4" textStyle="h4">
            Search for it
        </Text>
        <InputGroup>
            <Input
                placeholder="Germanshepherd.."
                value={searchValue}
                onChange={({ target }) => onSearch(target.value)}
                colorScheme="blue"
            />
            {searchValue.length > 0 && (
                <InputRightElement>
                    <CloseIcon _hover={{ cursor: 'pointer' }} data-testid="clearInput" onClick={clearSearch} />
                </InputRightElement>
            )}
        </InputGroup>
    </Box>
);

export default Search;
