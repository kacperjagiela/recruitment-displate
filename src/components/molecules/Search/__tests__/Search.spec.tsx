import React from 'react';
import { fireEvent, render, waitFor } from '@testing-library/react';

import Search from '../Search';

describe('<Search />', () => {
    it('renders correctly', () => {
        const { container } = render(<Search searchValue="test" onSearch={jest.fn} clearSearch={jest.fn} />);

        expect(container).toMatchSnapshot();
    });

    it('renders correctly with empty value', () => {
        const { container } = render(<Search searchValue="" onSearch={jest.fn} clearSearch={jest.fn} />);

        expect(container).toMatchSnapshot();
    });

    it('clears input by clicking X', () => {
        const clearSearch = jest.fn();
        const { getByPlaceholderText, getByTestId } = render(
            <Search searchValue="" onSearch={jest.fn} clearSearch={clearSearch} />,
        );

        const input = getByPlaceholderText('Germanshepherd..') as HTMLInputElement;

        waitFor(() => {
            fireEvent.change(input, { target: { value: 'test doggie' } });

            expect(getByTestId('clearInput')).toBeInTheDocument();

            fireEvent.click(getByTestId('clearInput'));

            expect(getByTestId('clearInput')).not.toBeInTheDocument();
            expect(input.value).toBe('');
        });
    });
});
