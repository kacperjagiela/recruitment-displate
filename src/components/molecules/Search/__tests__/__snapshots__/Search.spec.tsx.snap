// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> renders correctly 1`] = `
<div>
  <div
    class="css-18ws1q3"
  >
    <h2
      class="chakra-text css-0"
    >
      Woof Woof Gallery
    </h2>
    <h3
      class="chakra-text css-o5g2qp"
    >
      Already know the bread?
    </h3>
    <h4
      class="chakra-text css-0"
    >
      Search for it
    </h4>
    <div
      class="chakra-input__group css-4302v8"
    >
      <input
        class="chakra-input css-0"
        placeholder="Germanshepherd.."
        value="test"
      />
      <div
        class="chakra-input__right-element css-dp9r5f"
      >
        <svg
          class="chakra-icon css-u1i02k"
          data-testid="clearInput"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Search /> renders correctly with empty value 1`] = `
<div>
  <div
    class="css-18ws1q3"
  >
    <h2
      class="chakra-text css-0"
    >
      Woof Woof Gallery
    </h2>
    <h3
      class="chakra-text css-o5g2qp"
    >
      Already know the bread?
    </h3>
    <h4
      class="chakra-text css-0"
    >
      Search for it
    </h4>
    <div
      class="chakra-input__group css-4302v8"
    >
      <input
        class="chakra-input css-0"
        placeholder="Germanshepherd.."
        value=""
      />
    </div>
  </div>
</div>
`;
